contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -10228,3 +10268,61 @@ get_original_query_string(void)
 {
 	return original_query_string;
 }
+
+/*
+ * For the given expr query, replace the catalog_name with the new_catalog_name.
+ */
+void replace_schema_name(PLtsql_expr *expr, char* catalog_name, char* new_catalog_name)
+{
+	while(true)
+	{
+		if(strcasestr(expr->query, catalog_name))
+		{
+			char* match;
+			match = strcasestr(expr->query, catalog_name);
+			if (match != NULL)
+			{
+				size_t len = strlen(expr->query);
+				size_t n1 = match - expr->query;   // # bytes before the match
+				size_t n2 = strlen(catalog_name);      // # bytes in the pattern string
+				size_t n3 = strlen(new_catalog_name);  // # bytes in the replacement string
+				size_t n4 = len - n1 - n2;        // # bytes after the pattern in the source string
+				char *result = malloc(n1 + n3 + n4 + 1);
+				if (result != NULL) {
+					// copy the initial portion
+					memcpy(result, expr->query, n1);
+					// copy the replacement string
+					memcpy(result + n1, new_catalog_name, n3);
+					// copy the trailing bytes, including the null terminator
+					memcpy(result + n1 + n3, match + n2, n4 + 1);
+				}
+				expr->query = result;
```

Comment:
Can we improvise the performance of this loop? Right now it can take O(m*n) roughly.